test/mpi/distributed/matrix.cpp

```diff
@@ -51,131 +51,274 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+    using local_index_type =
+        typename std::tuple_element<1, decltype(
+                                           ValueLocalGlobalIndexType())>::type;
+    using global_index_type =
+        typename std::tuple_element<2, decltype(
+                                           ValueLocalGlobalIndexType())>::type;
+    using dist_mtx_type = gko::distributed::Matrix<value_type, local_index_type,
+                                                   global_index_type>;
+    using dist_vec_type = gko::distributed::Vector<value_type>;
+    using Partition =
+        gko::distributed::Partition<local_index_type, global_index_type>;
+    using matrix_data = gko::matrix_data<value_type, global_index_type>;
 
-    int get_transfer_count() const { return transfer_count_; }
 
-    static std::unique_ptr<HostToDeviceLogger> create(
-        std::shared_ptr<const gko::Executor> exec)
+    MatrixCreation()
+        : ref(gko::ReferenceExecutor::create()),
+          size{5, 5},
+          comm(gko::mpi::communicator(MPI_COMM_WORLD)),
+          mat_input{size,
+                    {{0, 1, 1},
+                     {0, 3, 2},
+                     {1, 1, 3},
+                     {1, 2, 4},
+                     {2, 1, 5},
+                     {2, 2, 6},
+                     {3, 3, 8},
+                     {3, 4, 7},
+                     {4, 0, 9},
+                     {4, 4, 10}}},
+          dist_input{{{size, {{0, 1, 1}, {0, 3, 2}, {1, 1, 3}, {1, 2, 4}}},
+                      {size, {{2, 1, 5}, {2, 2, 6}, {3, 3, 8}, {3, 4, 7}}},
+                      {size, {{4, 0, 9}, {4, 4, 10}}}}},
+          engine(42)
     {
-        return std::unique_ptr<HostToDeviceLogger>(
-            new HostToDeviceLogger(std::move(exec)));
+        init_executor(ref, exec, comm);
+
+
+        row_part = Partition::build_from_contiguous(
+            exec, gko::array<global_index_type>(
+                      exec, I<global_index_type>{0, 2, 4, 5}));
+        col_part = Partition::build_from_mapping(
+            exec,
+            gko::array<gko::distributed::comm_index_type>(
+                exec, I<gko::distributed::comm_index_type>{1, 1, 2, 0, 0}),
+            3);
+
+        dist_mat = dist_mtx_type::create(exec, comm);
     }
 
-protected:
-    explicit HostToDeviceLogger(std::shared_ptr<const gko::Executor> exec)
-        : gko::log::Logger(exec, gko::log::Logger::copy_started_mask)
-    {}
+    void SetUp() override { ASSERT_EQ(comm.size(), 3); }
 
-private:
-    mutable int transfer_count_ = 0;
+
+    std::shared_ptr<gko::ReferenceExecutor> ref;
+    std::shared_ptr<gko::EXEC_TYPE> exec;
+    gko::dim<2> size;
+    gko::mpi::communicator comm;
+    std::shared_ptr<Partition> row_part;
+    std::shared_ptr<Partition> col_part;
+
+    gko::matrix_data<value_type, global_index_type> mat_input;
+    std::array<matrix_data, 3> dist_input;
+
+    std::unique_ptr<dist_mtx_type> dist_mat;
+
+    std::default_random_engine engine;
```

Comment:
nit: you are missing a couple of header includes in this file (which are all transitively included, but some of them would be better to include explicitly):
- `array`
- `memory`
- `random`
- `ginkgo/core/base/array.hpp`
- `ginkgo/core/base/matrtix_data.hpp`